File: single-language-analysis/dataset-files/java/TestSmell044.java
// Original URL: https://github.com/henrytxz/Algo/blob/f0b186aa128ebd01835013f432b793593c4a0e6d/slowTests/not_quick/ArrayInversionTest.java#L22-L50

public class TestSmell044 {
    @Ignore //coz it's not quick todo: to review
    @Test
    public void debugSortAndCount() throws Exception {
        File file = new File("resources/IntegerArray.txt");
        try {
//            System.out.println(Integer.MAX_VALUE);
            List<Integer> list = MyArrayUtil.ListIntegerFromFile(file,94650);      //94680
            System.out.println("list size:"+list.size());

            Integer[] A = MyArrayUtil.ListIntegersToIntegerArray(list);

            Pair<Integer[],Integer> D_total = ArrayInversion.SortAndCount(A, A.length, 0, A.length-1);

            Integer[] D = D_total.getLeft();

            MyArrayUtil.checkArrayIntegerSorted(D);
            System.out.println("expect "+D_total.getRight()+" inversions");
            int expectedInversions = ArrayInversion.bruteForceCount(A);
//            System.out.println("expect "+expectedInversions+" inversions");
            Assert.assertEquals(expectedInversions, (int) D_total.getRight());

        } catch (FileNotFoundException e) {
            e.printStackTrace();  //To change body of catch statement use File | Settings | File Templates.
            Assert.fail();
        } catch (IOException e) {
            e.printStackTrace();
            Assert.fail();
        }
    }
}

File: single-language-analysis/dataset-files/python/test_smell_044.py
import pytest

class TestArrayInversion:
    @pytest.mark.skip(reason="coz it's not quick todo: to review")
    def test_smell_044(self):
        file = "resources/IntegerArray.txt"
        try:
            list_ = MyArrayUtil.list_integer_from_file(file, 94650)
            print("list size:" + str(len(list_)))

            A = MyArrayUtil.list_integers_to_integer_array(list_)

            D_total = ArrayInversion.sort_and_count(A, len(A), 0, len(A) - 1)

            D = D_total.get_left()

            MyArrayUtil.check_array_integer_sorted(D)
            print("expect " + str(D_total.get_right()) + " inversions")
            expected_inversions = ArrayInversion.brute_force_count(A)
            assert expected_inversions == D_total.get_right()

        except FileNotFoundError as e:
            print(e)
            assert False
        except IOError as e:
            print(e)
            assert False